tests/ui/methods.rs

```diff
@@ -689,3 +689,8 @@ fn temporary_cstring() {
 
 
 }
+
+fn iter_clone_collect() {
+    let v = [1,2,3,4,5];
+    let v2 : Vec<isize> = v.iter().cloned().collect();
+}
```

Comment:
Can you add a true-negative test with a conversion to a non-`Vec` as per https://github.com/Manishearth/rust-clippy/issues/1292#issuecomment-279205712?